src/main/CMakeLists.txt

```diff
@@ -36,16 +36,18 @@ set_source_files_properties(${libmain_gen_src_files} PROPERTIES GENERATED TRUE)
 
 add_library(main OBJECT ${libmain_src_files} ${libmain_gen_src_files})
 target_compile_definitions(main PRIVATE -D__BUILDING_CVC5DRIVER)
-if(ENABLE_SHARED)
-  set_target_properties(main PROPERTIES POSITION_INDEPENDENT_CODE ON)
+set_target_properties(main PROPERTIES POSITION_INDEPENDENT_CODE ON)
+
+if(ENABLE_STATIC_BINARY)
+  target_link_libraries(main PUBLIC cvc5-static cvc5parser-static)
+else()
+  target_link_libraries(main PUBLIC cvc5-shared cvc5parser-shared)
+endif()
+if(USE_CLN)
+  target_link_libraries(main PUBLIC CLN)
 endif()
 
-# We can't use target_link_libraries(main ...) here since this is only
-# supported for cmake version >= 3.12. Hence, we have to manually add the
-# library dependencies for main. As a consequence, include directories from
-# dependencies are not propagated and we need to manually add the include
-# directories of libcvc5 to main.
-add_dependencies(main cvc5 cvc5parser gen-tokens)
```

Comment:
Do we need to bump the cmake requirement to 3.12 now?

Reply:
To be honest, I lost track of which cmake version can build this now. My suggestion would be to simply try with a number of cmake versions to figure it out.

Reply:
The current state apparently works from cmake 3.15 upwards. This would support:
- ubuntu 20.04 (not 18.04)
- debian buster and stretch (i.e. oldstable and oldoldstable) with backports enabled
- debian bullseye (stable)
- homebrew

Reply:
Actually, it looks like it builds on Ubuntu 18.04 with CMake 3.10 (that's part of the CI).

Reply:
the CI image uses a newer cmake: https://github.com/actions/virtual-environments/blob/ubuntu18/20210919.1/images/linux/Ubuntu1804-README.md
It actually uses cmake 3.21.